src/main/java/org/opensearch/security/securityconf/ConfigModelV6.java

```diff
@@ -482,10 +482,20 @@ public boolean get(Resolved resolved, User user, String[] actions, IndexNameExpr
             return false;
         }
 
+        @Override
         public boolean impliesClusterPermissionPermission(String action) {
             return roles.stream().filter(r -> r.impliesClusterPermission(action)).count() > 0;
         }
 
+        @Override
+        public boolean strictClusterPermissionPermission(String action) {
+            return roles.stream()
+                    .map(r -> {
+                        final WildcardMatcher m = WildcardMatcher.from(r.clusterPerms);
+                        return m == WildcardMatcher.ANY ? WildcardMatcher.NONE : m;
```

Comment:
Is this return ternary condition backwards? Isn't this saying "if wildcardmatcher.matches(any) return "NONE" else return "m"" and don't you want it to be the other way around? 

I may be misinterpreting this statement.

Reply:
This way Im trying to avoid of such configueation:
```yaml
all_access:
  reserved: true
  hidden: false
  static: true
  description: "Allow full access to all indices and all cluster APIs"
  cluster_permissions:
    - "*"
``` 

so any user without `restapi:admin/*` permission does not have access to the endpoint. TBH I do not like it this way maybe better solution exists?

Reply:
Gotcha, that makes sense given what you are trying to do. Perhaps you could leave a short comment mentioning this?

Reply:
Sure will add.

Reply:
`admin` gets assigned to `all_access` by default here: https://github.com/opensearch-project/security/blob/main/config/roles_mapping.yml#L13-L17

This will mean that admin does not get the `restapi:*` permissions by default, right? If not, then the role would need to be updated or a new role created using the API, securityadmin or by creating a brand new security index in a new cluster. 

If `admin` currently has the ability to use the REST APIs, then after this change the `admin` should still be able to modify the securityconfig via API by default. I'm not positive of the behavior, but will follow-up after testing.

@willyborankin What do you think about this? `all_access` that gives all API access and a new role that has permissions `cluster:*` and `indices:*`. 

```
all_access:
  reserved: true
  hidden: false
  static: true
  description: "Allow full access to all APIs"
  cluster_permissions:
    - "*"

all_cluster_and_indices:
  reserved: true
  hidden: false
  static: true
  description: "Allow full access to all indices and all cluster APIs"
  cluster_permissions:
    - "cluster:*"
    - "indices:*"
```